infra/feast-operator/test/e2e/test_util.go

```diff
@@ -220,9 +220,13 @@ func isFeatureStoreHavingRemoteRegistry(namespace, featureStoreName string) (boo
 		return false, nil // Return false if "remote" is not present
 	}
 
+	// either remote hostname or feastRef should be available for remote config.
 	hostname, ok := remote["hostname"].(string)
 	if !ok || hostname == "" {
-		return false, nil // Return false if "hostname" is missing or empty
+		_, feastRefOk := remote["feastRef"].(map[string]interface{})
```

Comment:
i see you doing a lot of interface mapping with these tests, but i was curious if you've tried to unmarshall into an actual Go object type? e.g. `RemoteRegistryConfig` ?

Reply:
https://github.com/feast-dev/feast/blob/19424bcc975d90d922791b5bd0da6ac13955c0c5/infra/feast-operator/api/v1alpha1/featurestore_types.go#L266

Reply:
or you could unmarshall the full `FeatureStoreSpec` even after doing the initial `kubectl get`

Reply:
Initially i was overthinking not to have code dependency but i think it is ok to have dependency on the go lang struct objects from the same codebase. I have unmarshalled into `v1alpha1.Registry` object. 
 
 please take a look [at commit](https://github.com/feast-dev/feast/pull/4822/commits/7db4bf534f9c861226a60640bf07b730ad1f31c9).

Reply:
yes! that's great. its much more complex _not_ to leverage the existing go APIs. maybe in a followup PR this could be done throughout the e2e code?

Reply:
sure. i will make necessary changes in the next PR. I will leverage the existing go struct.